lib/modem_jwt/modem_jwt.c

```diff
@@ -93,71 +86,41 @@ int modem_jwt_generate(struct jwt_data *const jwt)
 		return -EMSGSIZE;
 	}
 
-	int ret = 0;
-	enum at_cmd_state state;
-	char *cmd;
-	char *cmd_resp = NULL;
-	size_t cmd_sz = sizeof(JWT_CMD_TEMPLATE_SEC_TAG) + JWT_CMD_PRINT_INT_SZ +
-			(jwt->subject ? strlen(jwt->subject) : 0) +
-			(jwt->audience ? strlen(jwt->audience) : 0);
+	int ret;
+	char *cmd_resp;
 
-	ret = at_cmd_init();
-	if (ret) {
-		return -EPIPE;
-	}
 
-	/* Allocate and format the JWT request cmd */
-	cmd = k_calloc(cmd_sz, 1);
-	if (!cmd) {
+	/* Allocate response buffer */
+	cmd_resp = k_calloc(CONFIG_MODEM_JWT_MAX_LEN, 1);
+	if (!cmd_resp) {
 		return -ENOMEM;
 	}
 
 	if (jwt->sec_tag) {
-		ret = snprintf(cmd, cmd_sz, JWT_CMD_TEMPLATE_SEC_TAG,
+		ret = nrf_modem_at_cmd(cmd_resp, CONFIG_MODEM_JWT_MAX_LEN,
+			JWT_CMD_TEMPLATE_SEC_TAG,
 			jwt->alg,
 			jwt->exp_delta_s,
 			jwt->subject ? jwt->subject : "",
 			jwt->audience ? jwt->audience : "",
 			jwt->sec_tag,
 			jwt->key);
 	} else {
-		ret = snprintf(cmd, cmd_sz, JWT_CMD_TEMPLATE,
+		ret = nrf_modem_at_cmd(cmd_resp, CONFIG_MODEM_JWT_MAX_LEN,
```

Comment:
`nrf_modem_at_cmd` can return a positive value, but `modem_jwt_generate` should not

Reply:
Fixed